.github/workflows/java_nightly.yml

```diff
@@ -60,60 +61,29 @@ jobs:
       - name: Install Archery
         shell: bash
         run: pip install -e arrow/dev/archery[all]
-#      - name: Download Artifacts
-#        run: |
-#          if [ -z $PREFIX ]; then
-#            PREFIX=nightly-packaging-$(date +%Y-%m-%d)-0
-#          fi
-#          echo $PREFIX
-#          # problem about rate limit using github3 without API_TOKEN
-#          archery crossbow download-artifacts -f java-jars -t arrow_java_$(date +"%d-%m-%Y") $PREFIX
-#          if [ -n "$(ls -A arrow_java_$(date +"%d-%m-%Y")/*)" ]; then
-#            echo "Found files!"
-#            echo "GOT_FILES=true" >> $GITHUB_ENV
-#          else
-#            echo "No files found. Stopping upload."
-#            echo "GOT_FILES=false" >> $GITHUB_ENV
-#          fi
-#         tree arrow_java_$(date +"%d-%m-%Y")
-      - shell: bash
-        name: Download Artifacts
+      - name: Download Artifacts
         run: |
           if [ -z $PREFIX ]; then
-            PREFIX=nightly-packaging-$(date +%Y-%m-%d)-0
-          fi
-          if [ -z $DEPENDENCY_TO_INSTALL ]; then
-            DEPENDENCY_TO_INSTALL=arrow
+            PREFIX=$(date +%Y-%m-%d)
           fi
-          ARROW_JAVA_NIGHTLY_VERSION=$PREFIX'-github-java-jars'
-          TMP_FOLDER=arrow_java_$(date +%Y-%m-%d)
-          mkdir -p $TMP_FOLDER
-          pushd $TMP_FOLDER
-          # this download artifacts should be executed thru crossbow (workaround to solve problem about rate limit using github3 without API_TOKEN)
-          echo "**************** 1 - Download $ARROW_JAVA_NIGHTLY_VERSION dependencies ****************"
-          wget $( \
-            wget \
-              "-qO-" "https://api.github.com/repos/ursacomputing/crossbow/releases/tags/$ARROW_JAVA_NIGHTLY_VERSION" \
-              | jq -r '.assets[] | select((.name | endswith(".pom")) or (.name | endswith(".jar"))) | .browser_download_url' \
-              | grep $DEPENDENCY_TO_INSTALL )
-          if [ -n "$(ls -A *)" ]; then
+          archery crossbow download-artifacts -f java-jars -t binaries nightly-packaging-$PREFIX-0
+          if [ -n "$(ls -A binaries/*/*/)" ]; then
             echo "Found files!"
             echo "GOT_FILES=true" >> $GITHUB_ENV
           else
             echo "No files found. Stopping upload."
             echo "GOT_FILES=false" >> $GITHUB_ENV
           fi
```

Comment:
I have changed this behaviour and removed the envvar in the R workflow in favour of explicitly failing the job with `exit 1` if there are no files downloaded. You can then also remove the `if: `s that reference it

Reply:
Changed